### PR TITLE
nvidia_x11: 470.94 → 470.103.01

### PR DESCRIPTION
**Pull request by smunix**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smunix** [comment]
@baracoder @nrdxp @fabianhjr 

/marvin opt-in
/status needs_reviewer

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 166065` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.nvidia_x11_legacy470</li>
    <li>linuxPackages_hardkernel_latest.nvidia_x11_legacy470</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy470</li>
    <li>linuxPackages.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_15.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_5_16.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy470</li>
    <li>linuxPackages_latest-libre.nvidia_x11_legacy470 (linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy470)</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11_legacy470</li>
    <li>linuxPackages_4_14_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_4_14_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages_4_19_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11_legacy470</li>
    <li>linuxPackages_5_10_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages_hardened.nvidia_x11_legacy470 (linuxPackages_5_15_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages_latest.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_17.nvidia_x11_legacy470)</li>
    <li>linuxPackages_5_4_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages-libre.nvidia_x11_legacy470 (linuxKernel.packages.linux_libre.nvidia_x11_legacy470)</li>
    <li>linuxPackages_lqx.nvidia_x11_legacy470 (linuxKernel.packages.linux_lqx.nvidia_x11_legacy470)</li>
    <li>linuxPackages_testing_bcachefs.nvidia_x11_legacy470 (linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_legacy470)</li>
    <li>linuxPackages_xanmod.nvidia_x11_legacy470 (linuxKernel.packages.linux_xanmod.nvidia_x11_legacy470)</li>
    <li>linuxPackages_zen.nvidia_x11_legacy470 (linuxKernel.packages.linux_zen.nvidia_x11_legacy470)</li>
  </ul>
</details>

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -54,9 +54,9 @@ rec {
 
   # Last one supporting Kepler architecture
   legacy_470 = generic {
-      version = "470.94";
-      sha256_64bit = "lYWqKTMOutm98izjyiusICbIWpoy8D18WfcUp3mFAOs=";
-      settingsSha256 = "blJNKuFu/Th/ceexkKhTH/eYk8miUlTT+ESrcIyJNn0=";
+      version = "470.103.01";
+      sha256_64bit = "VsIwn4nCE0Y7DEY2D3siddc3HTxyevP+3IjElu3Ih6U=";
+      settingsSha256 = "Roc2OFSNEnIGLVwP0D9f8vFTf5v3KkL99S0mZBWN7s0=";
       persistencedSha256 = "xnccQ/EgafwnReBlk5Y7iClAj4hwXyFq9gUmwqyEuwE=";
```
Comment (fabianhjr) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:60]: The persistencd hash should be updated too unless the version is overridden to remain at a fixed point (though not the case on this release)

**Review by fabianhjr — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/nvidia-x11/default.nix (hunk 1 above)
Comment (fabianhjr) [reply pkgs/os-specific/linux/nvidia-x11/default.nix:60]:
```
error: hash mismatch in fixed-output derivation '/nix/store/0c371867gdsqrix0igqw05ims9fg99j3-source.drv':
         specified: sha256-xnccQ/EgafwnReBlk5Y7iClAj4hwXyFq9gUmwqyEuwE=
            got:    sha256-AVI0j2bpfMCMBTKuQp+BoCewaXIW3Xt4NnV1fjAHOr0=
```

Comment (smunix) [reply pkgs/os-specific/linux/nvidia-x11/default.nix]: it's fixed now

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 166065` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxKernel.packages.hardkernel_4_14.nvidia_x11_legacy470</li>
    <li>linuxPackages_hardkernel_latest.nvidia_x11_legacy470</li>
  </ul>
</details>
<details>
  <summary>19 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_14.nvidia_x11_legacy470</li>
    <li>linuxPackages_4_14_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_4_14_hardened.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_legacy470</li>
    <li>linuxPackages_4_19_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_4_9.nvidia_x11_legacy470</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy470</li>
    <li>linuxPackages_5_10_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_15.nvidia_x11_legacy470)</li>
    <li>linuxPackages_hardened.nvidia_x11_legacy470 (linuxPackages_5_15_hardened.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_5_16.nvidia_x11_legacy470</li>
    <li>linuxPackages_latest.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_17.nvidia_x11_legacy470)</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy470</li>
    <li>linuxPackages_5_4_hardened.nvidia_x11_legacy470 (linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy470)</li>
    <li>linuxPackages_latest-libre.nvidia_x11_legacy470 (linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy470)</li>
    <li>linuxPackages-libre.nvidia_x11_legacy470 (linuxKernel.packages.linux_libre.nvidia_x11_legacy470)</li>
    <li>linuxPackages_lqx.nvidia_x11_legacy470 (linuxKernel.packages.linux_lqx.nvidia_x11_legacy470)</li>
    <li>linuxPackages_testing_bcachefs.nvidia_x11_legacy470 (linuxKernel.packages.linux_testing_bcachefs.nvidia_x11_legacy470)</li>
    <li>linuxPackages_xanmod.nvidia_x11_legacy470 (linuxKernel.packages.linux_xanmod.nvidia_x11_legacy470)</li>
    <li>linuxPackages_zen.nvidia_x11_legacy470 (linuxKernel.packages.linux_zen.nvidia_x11_legacy470)</li>
  </ul>
</details>

**Pull request merged**